bin/assigner/src/main.cpp

```diff
@@ -825,8 +1205,9 @@ int main(int argc, char *argv[]) {
             return 1;
         }
     }
+    gen_mode = gen_mode | nil::blueprint::generation_mode::assignments(); // for now in any case we generate assignment table to determine used_rows
 
-    if (!vm.count("public-input") && !vm.count("private-input") && gen_mode.has_assignments()) {
+    if (!vm.count("public-input") && !vm.count("private-input") && !gen_mode.has_size_estimation()) {
```

Comment:
No need inputs or generate circuits, we can estimate `used_rows` from signature function.
See how works `InputReader` if `has_value = false`